src/main/java/com/example/demo/services/DefaultPushNotificationsService.java

```diff
@@ -0,0 +1,191 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+public class DefaultPushNotificationsService implements PushNotificationsService{
+	
+	private static final long HALF_DAY_MILIS=12 * 60 * 60 * 1000;
+	private static final byte[] P256_HEAD= Base64.getDecoder().decode("MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgA");
+	
+	private ECPrivateKey privateKey;
+	private ECPublicKey publicKey;
+	
+	private PushSubsRepository pushSubsRepo;
+	private NotificationsRepository notificationsRepo;
+	
+	private PushMessageEncryptionUtil msgEncryptionUtil;
+	private HttpClient httpClient;
+	private Algorithm jwtAlgorithm;
+	
+	private ObjectMapper objectMapper;
+
+	public DefaultPushNotificationsService(PushSubsRepository pushSubsRepo, NotificationsRepository notificationsRepo, PushMessageEncryptionUtil msgEncryptionUtil,String privateKey,String publicKey) throws AuctionAppException {
+		this.pushSubsRepo=pushSubsRepo;
+		this.notificationsRepo=notificationsRepo;
+		
+		this.msgEncryptionUtil=msgEncryptionUtil;
+	    KeyFactory keyFactory=null;
+	    
+		try {
+			keyFactory = KeyFactory.getInstance("EC");
+		} catch (NoSuchAlgorithmException e) {
+			throw new BadInitializatinException();
+		}
+	    
+		try {
+			this.privateKey = (ECPrivateKey)keyFactory.generatePrivate(new PKCS8EncodedKeySpec(Base64.getDecoder().decode(privateKey.getBytes("UTF-8"))));
+			this.publicKey = (ECPublicKey)keyFactory.generatePublic(new X509EncodedKeySpec(Base64.getDecoder().decode(publicKey.getBytes("UTF-8"))));
+		} catch (UnsupportedEncodingException | InvalidKeySpecException e) {
+			throw new BadInitializatinException();
+		}
+        
+	    this.httpClient = HttpClient.newHttpClient();
+	    this.jwtAlgorithm = Algorithm.ECDSA256(this.publicKey, this.privateKey);
+	    this.objectMapper=new ObjectMapper();
+	}
+	
+	public byte[] getPublicKeyUncompressed() {
+	    byte[] result = new byte[65];
+	    byte[] encoded = publicKey.getEncoded();
+	    System.arraycopy(encoded, P256_HEAD.length, result, 0,
+	        encoded.length - P256_HEAD.length);
+	    return result;
+	}
+	
+	public String getPublicKeyBase64() {
+		return Base64.getUrlEncoder().withoutPadding().encodeToString(this.getPublicKeyUncompressed());
+	}
+	
+	public PushServiceSubModel subscribe(PushServiceSubModel model,User principal) {
+		PushSub entity=PushSub.fromModel(model);
+		entity.setUser(principal);
+		return pushSubsRepo.save(entity).toModel();
+	}
+	
+	public void notifyUser(User user,Notification notification) throws AuctionAppException {
+		notification.setUser(user);
+		notification.setTime(new Timestamp(System.currentTimeMillis()));
+		notificationsRepo.save(notification);
+		sendNotification(notification);
+	}
+	
+	public void sendMultipleNotifications(List<Notification> notifications){
+		notificationsRepo.saveAll(notifications);
+		notifications.stream().forEach(t -> {
+			try {
+				sendNotification(t);
+			} catch (AuctionAppException e) {}
```

Comment:
at least log this exception

Reply:
think when you would be on environment, a notification is not send, this way, you could not have fixed it without reproducint it locally, with logs you could get indication why process has failed